### DISCUSSION TITLE
TypeScript/NodeJS support

### DISCUSSION THREAD

**Discussion by ladderschool** [TimefoldAI/timefold-solver#947] [General]
Hello, what an amazing product timefold is. I was wondering if it would be possible to integrate it into a nodejs application that uses typescript on the frontend?

**Comment by triceo** [comment]
Hello @ladderschool! First of all, thank you for the feedback, we appreciate it.

To answer your question - we have only just released [Timefold Solver for Python](https://pypi.org/project/timefold/) and at the moment, we are not planning to integrate with any more languages. That said, if you stick with us for a bit longer, you may just see us deliver a hosted solution with a REST API, who knows. 😉 

In the meantime, the [quickstarts](https://github.com/TimefoldAI/timefold-quickstarts/) show you how to deliver such solution on your own, in the languages we already support: Java, Kotlin and Python. If you package that in a Docker container, nobody ever needs to know that there is Java underneath.
